### PR TITLE
add desired_capacity settings

### PR DESCRIPTION
**Pull request by dmitriy-lukyanchikov**
## what
- add desired_capacity settings

## why
- sometimes useful to set add `var.desired_capacity`

## references
* closes https://github.com/cloudposse/terraform-aws-ec2-autoscale-group/issues/29

### CODE REVIEW

**Comment by dmitriy-lukyanchikov** [comment]
looks like an issue with CI

@aknysh can you advise what to do here?

**Comment by aknysh** [comment]
we need to update the Terratest to `Go` modules.
Please run the following commands:

```
make init
make github/init
cd test/src/
go mod init github.com/cloudposse/$(cd ../../ && basename `pwd`)
go mod tidy
git rm Gopkg.*
git add go.*
cp ../../../terraform-null-label/test/src/Makefile 
```

**Comment by dmitriy-lukyanchikov** [comment]
ok its done, how can i rerun tests?

**Comment by Nuru** [comment]
@dmitriy-lukyanchikov After #39 merges, please rebase and update this PR. Then we can test it.

**Review by osterman — DISMISSED** [review]

File: variables.tf [hunk 1]
```diff
@@ -398,6 +398,12 @@ variable "cpu_utilization_low_statistic" {
   description = "The statistic to apply to the alarm's associated metric. Either of the following is supported: `SampleCount`, `Average`, `Sum`, `Minimum`, `Maximum`"
 }
 
+variable "desired_capacity" {
+  type        = number
+  description = "The number of Amazon EC2 instances that should be running in the group, if not set will use `min_size` as value"
+  default     = 0
```
Comment (nitrocode) [inline variables.tf:404]: A desired count of 0 is a bit unnerving to me. A safer option is to set it to `null`.

```suggestion
  default     = null
```

File: main.tf [hunk 2]
```diff
@@ -128,6 +128,7 @@ locals {
       launch_template        = local.launch_template_block
       override               = var.mixed_instances_policy.override
   })
+  desired_capacity = var.desired_capacity == 0 ? var.min_size : var.desired_capacity
```
Comment (nitrocode) [inline main.tf:131]: See other comment

```suggestion
  desired_capacity = var.desired_capacity == null ? var.min_size : var.desired_capacity
```

**Review by nitrocode — DISMISSED** [review]

**Review by jamengual — APPROVED** [review]

**Pull request merged**